app/src/androidTest/java/com/github/se/project/ui/overview/OverViewTest.kt

```diff
@@ -1,363 +1,398 @@
+
+  @Before
+  fun setup() {
+    /// Mock the dependencies
+    profilesRepository = mock(ProfilesRepository::class.java)
+    lessonRepository = mock(LessonRepository::class.java)
+    navController = mock()
+
+    // Initialize the ViewModel with a spy
+    listProfilesViewModel = ListProfilesViewModel(profilesRepository)
+    listProfilesViewModel = spy(listProfilesViewModel)
+
+    lessonViewModel = LessonViewModel(lessonRepository)
+    lessonViewModel = spy(lessonViewModel)
+
+    // Mock NetworkStatusViewModel
+    networkStatusViewModel =
+        object :
+            NetworkStatusViewModel(
+                application = androidx.test.core.app.ApplicationProvider.getApplicationContext()) {
+          override val isConnected = mockIsConnected
         }
-        composeTestRule.onNodeWithContentDescription("Profile Icon").performClick()
-        verify(navigationActions).navigateTo(anyString())
-    }
 
-    @Test
-    fun testLessonItemsDisplayed() {
-        composeTestRule.setContent {
-            HomeScreen(
-                listProfilesViewModel,
-                lessonViewModel,
-                networkStatusViewModel,
-                chatViewModel,
-                navigationActions)
-        }
-        composeTestRule.onNodeWithText("Physics Tutoring").assertIsDisplayed()
-    }
+    navigationActions = NavigationActions(navController)
+    navigationActions = spy(navigationActions)
 
-    @Test
-    fun testSectionDisplays() {
-        composeTestRule.setContent {
-            HomeScreen(
-                listProfilesViewModel,
-                lessonViewModel,
-                networkStatusViewModel,
-                chatViewModel,
-                navigationActions)
-        }
-        composeTestRule.onNodeWithText("Waiting for your Confirmation").assertIsDisplayed()
-        composeTestRule.onNodeWithText("Waiting for the Student Confirmation").assertIsDisplayed()
-        composeTestRule.onNodeWithText("Upcoming Lessons").assertIsDisplayed()
-        composeTestRule.onNodeWithText("Upcoming Instant Lesson").assertIsNotDisplayed()
-    }
+    // Stub the methods in NavigationActions
+    doNothing().`when`(navigationActions).navigateTo(anyString())
+    `when`(navigationActions.currentRoute()).thenReturn(Route.HOME)
 
-    @Test
-    fun testNoProfileFoundScreenDisplayed() {
-        // Set a null profile in the ViewModel
-        listProfilesViewModel =
-            mock(ListProfilesViewModel::class.java).apply {
-                `when`(currentProfile).thenReturn(MutableStateFlow(null))
-            }
-        // Recompose
-        composeTestRule.setContent {
-            HomeScreen(
-                listProfilesViewModel,
-                lessonViewModel,
-                networkStatusViewModel,
-                chatViewModel,
-                navigationActions)
-        }
+    // Stub the methods in ChatViewModel
+    // doNothing().`when`(chatViewModel).connectUser(any())
```

Comment:
Why are these commented out? If they're not needed, they would be better off being removed